### PR TITLE
Add com.rosegardenmusic.rosegarden

### PR DESCRIPTION
**Pull request by hfiguiere**

### CODE REVIEW

**Comment by nedrichards** [comment]
bot, buld com.rosegardenmusic.rosegarden
testbot, build com.rosegardenmusic.rosegarden

File: com.rosegardenmusic.rosegarden.json [hunk 1]
```diff
@@ -0,0 +1,177 @@
+{
+  "app-id": "com.rosegardenmusic.rosegarden",
+  "runtime": "org.kde.Platform",
+  "runtime-version": "5.11",
```
Comment (barthalion) [inline com.rosegardenmusic.rosegarden.json]: As in the other PRs, let's try to make it work on 5.12.

**Comment by eszlari** [comment]
Are you sure about the `app-id`? Shouldn't the name be capitalized: `com.rosegardenmusic.Rosegarden`

Maybe you should contact upstream about it.

**Comment by hfiguiere** [comment]
They have an appdata file that uses `com.rosegardenmusic.rosegarden.desktop` as the ID.

**Comment by nedrichards** [comment]
Then we should really use `com.rosegardenmusic.rosegarden.desktop`

**Comment by nedrichards** [comment]
bot, build com.rosegardenmusic.Rosegarden
testbot, build com.rosegardenmusic.Rosegarden

**Comment by barthalion** [comment]
LGTM , but please change app-id to match one used by upstream (`com.rosegardenmusic.rosegarden`).

File: com.rosegardenmusic.Rosegarden.json [hunk 2]
```diff
@@ -0,0 +1,191 @@
+{
+  "app-id": "com.rosegardenmusic.Rosegarden",
+  "runtime": "org.kde.Platform",
+  "runtime-version": "5.12",
+  "sdk": "org.kde.Sdk",
+  "command": "rosegarden",
+  "rename-appdata-file": "rosegarden.appdata.xml",
```
Comment (barthalion) [inline com.rosegardenmusic.Rosegarden.json]: Ideally it should be upstreamed.

**Comment by barthalion** [comment]
Also, have you contacted upstream team if they would like to help?

**Comment by hfiguiere** [comment]
We should mark this as blocked (like Ardour). Sound playback requires Jack.

**Comment by nedrichards** [comment]
bot, build com.rosegardenmusic.Rosegarden
testbot, build com.rosegardenmusic.Rosegarden

**Comment by hfiguiere** [comment]
app-id was changed and I forgot to change the title...

**Comment by nedrichards** [comment]
bot, build com.rosegardenmusic.rosegarden

**Comment by hfiguiere** [comment]
dssi fails on aarch64 because of configure.... will look into it.

**Comment by nedrichards** [comment]
bot, build com.rosegardenmusic.rosegarden

**Comment by nedrichards** [comment]
bot, build com.rosegardenmusic.rosegarden

**Comment by nedrichards** [comment]
The blocker now is that the appdata doesn't comply with the flathub requirements (we need an OARS rating an release tags).

**Comment by nedrichards** [comment]
Also, as [noted by @barthalion upthread](https://github.com/flathub/flathub/pull/821#issuecomment-462255477) have you spoken to upstream about using flathub? If there's a mailing list post or bug then chucking a link here would be very helpful.

**Comment by hfiguiere** [comment]
bot, build com.rosegarden.rosegarden

**Comment by hfiguiere** [comment]
bot, build com.rosegardenmusic.rosegarden

**Comment by hfiguiere** [comment]
new blocker: the jack2 hack to start a local server fail because it requires lsof in the alsa driver

https://github.com/jackaudio/jack2/blob/4341f8086f795bdbb3203aa3046098ddef2f6651/linux/alsa/alsa_driver.c#L1896

lsof is gone from the SDK

**Comment by hfiguiere** [comment]
appdata needs fixing now.

**Comment by hfiguiere** [comment]
bot, build com.rosegardenmusic.rosegarden

**Comment by hfiguiere** [comment]
Patch submitted upstream https://sourceforge.net/p/rosegarden/patches/69/

**Comment by hfiguiere** [comment]
Queued test build for com.rosegardenmusic.rosegarden

**Comment by hfiguiere** [comment]
bot, build com.rosegardenmusic.rosegarden

**Comment by barthalion** [comment]
Still blocked due to Jack?

**Comment by hfiguiere** [comment]
The "jack hack" (ie starting jackd) was somewhat working, but the new SDK broke it. I have something in the way but they haven't been pushed yet.

FWIW, Pipewire is not a short term solution for jack support - it has a long way to go from what the developers are planning.

**Comment by nedrichards** [comment]
bot, build com.rosegardenmusic.rosegarden

**Comment by nedrichards** [comment]
bot, build com.rosegardenmusic.rosegarden

**Comment by hfiguiere** [comment]
bot, build com.rosegardenmusic.rosegarden

**Comment by LeandroStanger** [comment]
bot, build com.rosegardenmusic.rosegarden

**Comment by hfiguiere** [comment]
This doesn't change the status of jack not working.

**Comment by hfiguiere** [comment]
bot, build com.rosegardenmusic.rosegarden

**Comment by hfiguiere** [comment]
I still can't figure out how to have pipewire output sound, but when I do, it should work.

**Comment by hfiguiere** [comment]
bot, build com.rosegardenmusic.rosegarden

**Comment by hfiguiere** [comment]
Works with Pipewire for JACK support.

**Comment by barthalion** [comment]
Not sure if anyone has asked but have you tried to contact upstream maintainers about this submission?

**Comment by hfiguiere** [comment]
> Not sure if anyone has asked but have you tried to contact upstream maintainers about this submission?

Request upstream: https://sourceforge.net/p/rosegarden/feature-requests/488/

**Comment by fedelibre** [comment]
Blocked because you're waiting for answers from upstream?

I've built the flatpak locally and I'm trying to make it work on a system (Fedora 33) using PipeWire.
I cannot hear any sound when I play an imported MIDI file.

```
$ pw-jack flatpak run com.rosegardenmusic.rosegarden
Gtk-Message: 23:20:34.977: Failed to load module "canberra-gtk-module"
Gtk-Message: 23:20:34.977: Failed to load module "pk-gtk-module"
Gtk-Message: 23:20:34.978: Failed to load module "canberra-gtk-module"
Gtk-Message: 23:20:34.978: Failed to load module "pk-gtk-module"
Qt: Session management error: Could not open network socket
[ActionFileClient] WARNING: ActionFileClient( "App" )::findAction(): No such action as  "delete"
[DecoyAction] getInstance(): WARNING: Using decoy action
[ActionFileClient] WARNING: ActionFileClient( "App" )::findAction(): No such action as  "edit_cut"
[DecoyAction] getInstance(): WARNING: Using decoy action
[ActionFileClient] WARNING: ActionFileClient( "App" )::findAction(): No such action as  "rescale"
[DecoyAction] getInstance(): WARNING: Using decoy action
[ActionFileClient] WARNING: ActionFileClient( "App" )::findAction(): No such action as  "auto_split"
[DecoyAction] getInstance(): WARNING: Using decoy action
[ActionFileClient] WARNING: ActionFileClient( "App" )::findAction(): No such action as  "split_by_pitch"
[DecoyAction] getInstance(): WARNING: Using decoy action
[ActionFileClient] WARNING: ActionFileClient( "App" )::findAction(): No such action as  "split_by_recording"
[DecoyAction] getInstance(): WARNING: Using decoy action
[ActionFileClient] WARNING: ActionFileClient( "App" )::findAction(): No such action as  "split_at_time"
[DecoyAction] getInstance(): WARNING: Using decoy action
[ActionFileClient] WARNING: ActionFileClient( "App" )::findAction(): No such action as  "split_by_drum"
[DecoyAction] getInstance(): WARNING: Using decoy action
[ActionFileClient] WARNING: ActionFileClient( "App" )::findAction(): No such action as  "join_segments"
[DecoyAction] getInstance(): WARNING: Using decoy action
[ActionFileClient] WARNING: ActionFileClient( "App" )::findAction(): No such action as  "cut_range"
[DecoyAction] getInstance(): WARNING: Using decoy action
[W][000004347.889664][module-rtkit.c:499 idle_func()] could not make thread realtime: Errore di input/output
[W][000004347.894971][pipewire-jack.c:1783 client_node_port_use_buffers()] jack-client 0x7f6640006f30: Failed to mlock memory 0x7f665c17d080 32768: This is not a problem but for best performance, consider increasing RLIMIT_MEMLOCK
[W][000004347.896964][pipewire-jack.c:1783 client_node_port_use_buffers()] jack-client 0x7f6640006f30: Failed to mlock memory 0x7f665c13c040 32768: This is not a problem but for best performance, consider increasing RLIMIT_MEMLOCK
[W][000004347.896992][pipewire-jack.c:1783 client_node_port_use_buffers()] jack-client 0x7f6640006f30: Failed to mlock memory 0x7f665c133080 32768: This is not a problem but for best performance, consider increasing RLIMIT_MEMLOCK
[W][000004347.897015][pipewire-jack.c:1783 client_node_port_use_buffers()] jack-client 0x7f6640006f30: Failed to mlock memory 0x7f665c12a040 32768: This is not a problem but for best performance, consider increasing RLIMIT_MEMLOCK
[W][000004347.897485][pipewire-jack.c:1783 client_node_port_use_buffers()] jack-client 0x7f6640006f30: Failed to mlock memory 0x7f665c120040 32768: This is not a problem but for best performance, consider increasing RLIMIT_MEMLOCK
[Composition] getTrackById( 0 ): WARNING: Track ID not found.
[Composition]   Available track ids are:
[AlsaDriver] renameDevice(): WARNING: Cannot find device  1  in port map
[StartupTester] StartupTester::slotNetworkFinished(): Connection failed:  "Host www.rosegardenmusic.com non trovato"
[NoteFontMap] startElement(): WARNING: Unable to load any of the fonts in "fughetta"
[NoteFontMap] startElement(): WARNING: Unable to load any of the fonts in "inkpen2,inkpen"
[NoteFontMap] startElement(): WARNING: Unable to load any of the fonts in "maestro"
[NoteFontMap] startElement(): WARNING: Unable to load any of the fonts in "opus"
[NoteFontMap] startElement(): WARNING: Unable to load any of the fonts in "petrucci"
[NoteFontMap] startElement(): WARNING: Unable to load any of the fonts in "sonata"
[NoteFontMap] startElement(): WARNING: Unable to load any of the fonts in "steinberg notation"
[NoteFontMap] startElement(): WARNING: Unable to load any of the fonts in "xinfonia"
[Composition] getTrackById( 0 ): WARNING: Track ID not found.
[Composition]   Available track ids are:
[AlsaDriver] renameDevice(): WARNING: Cannot find device  1  in port map
Warning: Composition::~Composition() with 3 observers still extant
Observers are: 0x560ce22aa4d8 [N10Rosegarden17TrackParameterBoxE] 0x560ce21ed270 [N10Rosegarden20CompositionModelImplE] 0x560ce292efd0 [N10Rosegarden12TrackButtonsE]
```

A PipeWire bug maybe?

pw-jack is necessary?

**Comment by hfiguiere** [comment]
It was marked as blocked back then because it needs JACK. But now it works. However I can't remove the label @barthalion added.

As for the audio out, you probably need to connect the output using qjack or cadence (or anything that manage the JACK connections). There is an issue with pw-jack that the system interface don't have the same name, and I talked to the pipewire developers and was told it would be fixed.

See https://gitlab.freedesktop.org/pipewire/pipewire/-/issues/446

**Comment by hfiguiere** [comment]
It's a packaging bug IMHO. But I have the latter, and run qjackctl with pw-jack.

**Pull request closed without merging**